src/jit/lclvars.cpp

```diff
@@ -5760,6 +5761,21 @@ void Compiler::lvaAlignFrame()
             lvaIncrementFrameSize(sizeof(void*));
         }
     }
+#else
+    // The stack on x86 not Win32 must be 16 byte aligned.
+    if ((compLclFrameSize % STACK_ALIGN) != 0)
+    {
+        lvaIncrementFrameSize(STACK_ALIGN - (compLclFrameSize % STACK_ALIGN));
+    }
+    else if (lvaDoneFrameLayout != FINAL_FRAME_LAYOUT)
+    {
+        // If we are not doing final layout, we don't know the exact value of compLclFrameSize
+        // and thus do not know how much we will need to add in order to be aligned.
+        // We add 16 so compLclFrameSize is still a multiple of 16.
+        lvaIncrementFrameSize(STACK_ALIGN);
+    }
+    assert((compLclFrameSize % STACK_ALIGN) == 0);
```

Comment:
I think that you his need to be written differently:
First if it isn't the final frame layout phase you have to add on the maximum pad of 12 bytes.
Then after that you have to align compLclFrameSize  to a multiple of 16.
This can probably be unconditionally as well:

```
if (STACK_ALIGN >  REGSIZE_BYTES)
{
   // If we are not doing final layout, we don't know the exact value of compLclFrameSize  
   // and thus do not know how much we will need to add in order to be aligned.  
   // We add the maximum pad that we could ever have (which is 12)
   lvaIncrementFrameSize(STACK_ALIGN - REGSIZE_BYTES);

   // The stack must be 16 byte aligned.  
   if ((compLclFrameSize % STACK_ALIGN) != 0)  
   {  
       lvaIncrementFrameSize(STACK_ALIGN - (compLclFrameSize % STACK_ALIGN));  
   }
```




}